src/parse.c

```diff
@@ -1043,12 +1043,36 @@ handle_access_point_band(NetplanParser* npp, yaml_node_t* node, const void* _, G
     return TRUE;
 }
 
+static gboolean
+handle_wireguard_private_key_flags(NetplanParser* npp, yaml_node_t* node, const void* _, GError** error)
```

Comment:
thought: Maybe we can move some of the logic into a `handle_generic_key_flags` function, so we can handle wireguard-private-key-flags and wifi-psk-flags alike.